src/main.rs

```diff
@@ -797,15 +808,30 @@ impl fmt::Display for ScoperBuildError {
 impl Error for ScoperBuildError {}
 
 #[allow(clippy::cognitive_complexity)] // 🤷‍♀️ macros
-fn get_language_scopers(args: &cli::Cli) -> Vec<Box<dyn LanguageScoper>> {
+fn get_language_scopers(args: &cli::Cli) -> Result<Vec<Box<dyn LanguageScoper>>, ProgramError> {
+    fn read_query_file(query_or_path: &cli::CodeQuery, buf: &mut String) -> io::Result<Option<()>> {
```

Comment:
Allocating inside of `read_query_file` and returning results (if found) should be totally fine. This only runs at startup, so performance is not an issue. This might clean up this functions usage (i.e. allows pattern matching over `is_some` further down IIUC)

Reply:
Fixed in https://github.com/alexpovel/srgn/pull/144/commits/a9e1a7771b44b791c161da31d98dc7eea4aeb5e1